src/components/Date/Date.jsx

```diff
@@ -28,7 +27,7 @@ class Date extends React.Component {
   }
 
   UNSAFE_componentWillMount() {
-    this.id = _.uniqueId('date-input-');
+    this.inputId = _.uniqueId('date-input-');
```

Comment:
Changed this to `inputId` since the `errorSpanId` is using this name instead of `id` (line 87) - found when writing unit tests and the `errorSpanId` included `undefined` in the id.